### PR TITLE
add cli flag for to-resource

### PR DESCRIPTION
**Pull request by ryanfkeepers**
adds a cli flag for restoring to a different resource (mailbox, user, site) from the one stored in the
backup.

---

#### Does this PR need a docs update or release note?

- [x] :white_check_mark: Yes, it's included

#### Type of change

- [x] :sunflower: Feature

#### Issue(s)

* #3562

#### Test Plan

- [x] :muscle: Manual
- [x] :zap: Unit test

### CODE REVIEW

**Comment by ryanfkeepers** [comment]
@gmatev added you to the review for checks on flag names and documentation.

File: website/docs/setup/restore-options.md [hunk 1]
```diff
@@ -108,3 +108,18 @@ the copy of`reports.txt` is named `reports 1.txt`.
...
 version. If multiple existing items collide with the backup item, only one of the
 existing items is replaced.
+
+## To resource
+
+The `--to-resource` flag lets you select which resource will receive the restored data.
+A resource can be a mailbox, user, or sharepoint site.  
+
+<CodeBlock language="bash">{
+    `corso restore onedrive --backup abcd --to-resource adelev@alcion.ai`
```
Comment (pandeyabs) [inline website/docs/setup/restore-options.md:118]: Suggest changing `abcd` to a uuid format instead

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,8 +7,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased] (beta)
 
+### Added
+- Restore commands now accept an optional resource override with the `--to-resource` flag.  This allows restores to recreate backup data witthin different mailboxes, sites, and users.
```
Comment (pandeyabs) [inline CHANGELOG.md:11]:
```suggestion
- Restore commands now accept an optional resource override with the `--to-resource` flag.  This allows restores to recreate backup data within different mailboxes, sites, and users.  
```

File: src/cli/flags/restore_config.go [hunk 3]
```diff
@@ -25,5 +27,8 @@ func AddRestoreConfigFlags(cmd *cobra.Command) {
 		"Sets the behavior for existing item collisions: "+string(control.Skip)+", "+string(control.Copy)+", or "+string(control.Replace))
 	fs.StringVar(
 		&DestinationFV, DestinationFN, "",
-		"Overrides the destination where items get restored; '/' places items into their original location")
+		"Overrides the folder where items get restored; '/' places items into their original location")
+	fs.StringVar(
+		&ToResourceFV, ToResourceFN, "",
+		"Overrides the protected resource (mailbox, site, etc) where data gets restored")
```
Comment (pandeyabs) [inline src/cli/flags/restore_config.go:33]:
```suggestion
		"Overrides the protected resource (mailbox, site, user, etc) where data gets restored")
```

**Review by pandeyabs — APPROVED** [review]
Looks good! Left some nits

File: website/docs/setup/restore-options.md (hunk 1 above)
Comment (ryanfkeepers) [reply website/docs/setup/restore-options.md:118]: Going to keep this, as it currently matches our other examples.

File: src/cli/flags/restore_config.go (hunk 3 above)
Comment (ryanfkeepers) [reply src/cli/flags/restore_config.go:33]: Isn't that covered in `etc`? 😆

Comment (pandeyabs) [reply src/cli/flags/restore_config.go:33]: IMO, it's important enough to not be in `etc`

File: website/docs/setup/restore-options.md (hunk 1 above)
Comment (gmatev) [reply website/docs/setup/restore-options.md:118]: The structure of this page is a little off beyond just this flag.
* The Item collision handling sub options probably should move to H3 `###` under `## Item collision handling`
* The heading `To resource` while describing the flag correctly is probably not descriptive enough for a heading. Better option is probably `Restore resource target`. For consistency you could rename `Destination` above to `Restore folder target` or `Restore folder destination`
* Probably need to cover the interactions of `to-resource` with `destination`, if any. Assuming that the example restores to an alternate resource in a folder `Corso_Restore_<>` this should probably be mentioned for clarity. If we support combing the parameters, an extra example can be useful as well

Not sure, if related to the functionality here, but some of the behavior around restoring to a resource that may have been deleted at some point for `original location` should be covered somewhere in the docs.

**Review by gmatev — APPROVED** [review]
Ideally reorganize the restore options page a bit based on the feedback. Otherwise LGTM

File: website/docs/setup/restore-options.md (hunk 1 above)
Comment (ryanfkeepers) [reply website/docs/setup/restore-options.md:118]: fixed the suggested comments.

> some of the behavior around restoring to a resource that may have been deleted at some point for original location should be covered somewhere in the docs

There's the `limitation` at the bottom which states we won't restore to a missing or deleted resource.  Is that sufficient?

**Pull request merged**